utils/outputwriter/outputcontent.go

```diff
@@ -111,295 +150,705 @@ func untitledForJasMsg(writer OutputWriter) string {
+	// Hide optional columns if all empty (violations/no status)
+	table.GetColumnInfo("ID").OmitEmpty = true
+	table.GetColumnInfo("Status").OmitEmpty = true
+	table.GetColumnInfo("Watch Name").OmitEmpty = true
+	table.GetColumnInfo("Policies").OmitEmpty = true
+	// Construct rows
+	for _, issue := range issues {
+		// Determine the issue applicable status
+		applicability := jasutils.Applicable.String()
+		status := ""
+		if issue.Applicability != nil && issue.Applicability.Status != "" {
+			status = issue.Applicability.Status
+			if status == jasutils.Inactive.String() {
+				// Update the applicability status to Not Applicable for Inactive
+				applicability = jasutils.NotApplicable.String()
+			}
+		}
+		table.AddRowWithCellData(
+			NewCellData(writer.FormattedSeverity(issue.Severity, applicability)),
+			NewCellData(issue.IssueId),
+			NewCellData(status),
+			NewCellData(issue.Finding),
+			NewCellData(issue.Watch),
+			NewCellData(issue.Policies...),
+		)
 	}
-	return contentBuilder.String()
+	return table.Build()
+}
+
+func getSecretsRuleFullDescriptionTable(info formats.ScannerInfo, writer OutputWriter) *MarkdownTableBuilder {
+	table := getBaseJasDetailsTable(info, writer)
+	table.AddRow(MarkAsBold("Abbreviation:"), info.RuleId)
+	return table
 }
 
-func getJasDescriptionTable(severity, finding string, writer OutputWriter) string {
-	return NewMarkdownTable("Severity", "Finding").AddRow(writer.FormattedSeverity(severity, jasutils.Applicable.String()), finding).Build()
+// Utilities
+
+func getIssueType(violation bool) string {
+	if violation {
+		return "Violation"
+	}
+	return "Vulnerability"
 }
 
-func getSecretsDescriptionTable(severity, finding, status string, writer OutputWriter) string {
-	columns := []string{"Severity", "Finding"}
-	applicability := jasutils.Applicable.String()
-	if status != "" {
-		columns = append(columns, "Status")
-		if status == jasutils.Inactive.String() {
-			applicability = jasutils.NotApplicable.String()
+func getDirectDependenciesCellData(components []formats.ComponentRow) (dependencies CellData) {
+	if len(components) == 0 {
+		return NewCellData()
+	}
+	for _, component := range components {
+		dependencies = append(dependencies, results.GetDependencyId(component.Name, component.Version))
+	}
+	return
+}
+
+func getCveIdsCellData(cveRows []formats.CveRow, issueId string) (ids CellData) {
+	if len(cveRows) == 0 {
+		return NewCellData(issueId)
+	}
+	for _, cve := range cveRows {
+		ids = append(ids, cve.Id)
+	}
+	return
+}
+
+func getScaSecurityIssueDetailsContent(issues []formats.VulnerabilityOrViolationRow, violations bool, writer OutputWriter) (content []string) {
+	issuesWithDetails := getIssuesWithDetails(issues)
+	if len(issuesWithDetails) == 0 {
+		return
+	}
+	for _, issue := range issuesWithDetails {
+		if len(issues) == 1 {
+			content = append(content, getScaSecurityIssueDetails(issue, violations, writer))
+		} else {
+			content = append(content, "\n"+writer.MarkAsDetails(
```

Comment:
please extract the inner calls to variables and only then place them in the func. It is extreamly hard to read what is going on here. Id rather it to be longer but so we can understand it next time we come back to it